tests/QueryTest.php

```diff
@@ -1535,6 +1535,22 @@ public function testNestedOrAnd()
         );
     }
 
+    public function testNestedOrAndHaving()
+    {
+        $q = $this->q();
+        $q->table('employee')->field(new Expression('sum([])', ['amount']), 'salary')->group('type');
+        $q->having(
+            $q
+                ->orExpr()
+                ->having('a', 1)
```

Comment:
are we safe if `where()` is used here?

Reply:
Yes, even it has priority

Reply:
I think we should throw instead - when rendering parent where/having, check if the subexpression has not other kind set.

Reply:
and probably as another test method like `testNestedOrAndHavingWithWhereException`

Reply:
Why don't allow to mix `where` and `having`? In SQL it's possible to have:
```
SELECT name, sum(amount) total
FROM t
WHERE deleted=0
GROUP BY name
HAVING total > 100
```

Reply:
Only when nested! If however that what you have written is not allowed, then add a test and @georgehristov must fix asap :)

Reply:
@DarkSide666 this is only for the query nested expressions `Query::orExpr()` / `Query::andExpr()` used for building complex WHERE / HAVING. 
These use simple template `[orwhere]` or `[andwhere]` and only implode the conditions in a string with `or` / `and`

In the full query you can of course still use WHERE and HAVING combined.

Reply:
It would be great if we can rename creator methods like `orExpr` to `createOrExpr`.